### PR TITLE
Issue 50 - what I have so far for MessageEvents

### PR DESCRIPTION
**Pull request by tyrocca**
Hi @richleland. The message event thing  seemed like it only needed the get method to be implemented so I did that. I was able to pass most all the tests; however, the requests library was giving me problems. I'm not too familiar with python tests so I might have just been doing something wrong, but I know that when used with the proper details it works.

### CODE REVIEW

**Comment by richleland** [comment]
Thanks for submitting this @tyrocca! I'll take a look in the next couple of days for sure.

File: sparkpost/message_events.py [hunk 1]
```diff
@@ -0,0 +1,133 @@
...
+                param=param,
+                filter_val=filter_val
+            ))
+
+        request_str = "&".join(request_list)
+        return request_str
+
+    def _fetch_get(self, request_str):
+        uri = "%s?%s" % (self.uri, request_str)
+        results = self.request('GET', uri, return_all_keys=True)
```
Comment (richleland) [inline sparkpost/message_events.py]: I'm all for including the `return_all_keys` flag. Can you talk more about the use case behind setting this to `True` here? My concern is having a totally different payload coming through for this API. I wonder if we should make this an options that passes through as an arg on the `get` method.

File: sparkpost/message_events.py [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+            'campaign_ids': ('listable', None),
+            'events': ('listable', None),
+            'friendly_froms': ('listable', None),
+            'message_ids': ('listable', None),
+            'recipients': ('listable', None),
+            'subaccounts': ('listable', None),
+            'template_ids': ('listable', None),
+            'transmission_ids': ('listable', None),
+            # datetimes
+            'from': ('datetime', None),
```
Comment (richleland) [inline sparkpost/message_events.py]: `from` isn't usable as a keyword arg. You get this error:

```python
In [1]: sp.message_events.get(from='2016-04-01T00:00')
  File "<ipython-input-1-3001141e742e>", line 1
    sp.message_events.get(from='2016-04-01T00:00')
                             ^
SyntaxError: invalid syntax
```

File: test/test_message_events.py [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    }
+    formatted_kwargs = mesg._format_request_params(**kwargs)
+    assert "to=now&campaign_ids=1,2&per_page=100" == formatted_kwargs
+
+
+@responses.activate
+def test_fail_get():
+    responses.add(
+        responses.GET,
+        url="""https://api.sparkpost.com/api/v1/message-events?transmission_ids=8675309""",  # noqa
```
Comment (richleland) [inline test/test_message_events.py]: This test fails locally - you can remove the `?transmission_ids=8675309` part and it'll pass - reponses doesn't require a match on query params.

File: test/test_message_events.py [hunk 4]
```diff
@@ -0,0 +1,111 @@
...
+    with pytest.raises(SparkPostAPIException):
+        sp = SparkPost('fake-key')
+        sp.message_events.get(transmission_ids=8675309)
+
+
+@responses.activate
+def test_success_get():
+    responses.add(
+        responses.GET,
+        url="""https://api.sparkpost.com/api/v1/message-events?transmission_ids=8675309""",  # noqa
```
Comment (richleland) [inline test/test_message_events.py]: This test fails locally - you can remove the `?transmission_ids=8675309` part and it'll pass - reponses doesn't require a match on query params.

File: test/test_message_events.py [hunk 5]
```diff
@@ -0,0 +1,111 @@
+import responses
+import pytest
```
Comment (richleland) [inline test/test_message_events.py]: insert empty line after this one

File: test/test_message_events.py [hunk 6]
```diff
@@ -0,0 +1,111 @@
...
+from sparkpost import (
```
Comment (richleland) [inline test/test_message_events.py]: we can just put this on one line separated by commas

**Comment by richleland** [comment]
@tyrocca thanks for the hard work on this. I left a few comments on the PR - should be pretty simple to get those tests passing, and this has 100% coverage when they do so we're good on that front. Take a look at my comments and let me know if you have any questions/comments.

File: sparkpost/message_events.py (hunk 1 above)
Comment (tyrocca) [inline sparkpost/message_events.py]: That sounds like a good idea to me - I'll do that

**Comment by richleland** [comment]
Hi @tyrocca I'm going to go ahead and close this out, but feel free to re-submit if you've wrapped this up/fixed tests. Thanks for the contribution.

**Pull request closed without merging**